File: Leetcode/Daily/07-2025/TypeScript/03.ts
// Problem: 3304. Find the K-th Character in String Game I (Easy)
// Link: https://leetcode.com/problems/find-the-k-th-character-in-string-game-i

function kthCharacter(k: number): string {
    let add = -1;
    while (k % 2 === 0) {
        add++;
        k = (k / 2) | 0;
    }

    while (k > 0) {
        if (k % 2 === 1) add++;
        k = ~~(k / 2);
    }

    return String.fromCharCode( (add % 26) + 'a'.charCodeAt(0));
};

File: Leetcode/Daily/07-2025/Python3/03.py
# Problem: 3304. Find the K-th Character in String Game I (Easy)
# Link: https://leetcode.com/problems/find-the-k-th-character-in-string-game-i

class Solution:
    def kthCharacter(self, k: int) -> str:
        add = -1
        while k % 2 == 0:
            k //= 2
            add += 1

        while k > 0:
            if k % 2 == 1:
                add += 1
            k //= 2
        
        return chr((add % 26) + ord('a'))